cmd/nitro/init.go

```diff
@@ -602,6 +602,17 @@ func openInitializeChainDb(ctx context.Context, stack *node.Node, config *NodeCo
 		if err != nil {
 			return nil, nil, fmt.Errorf("couln't extract init archive '%v' err:%w", initFile, err)
 		}
+		wasmDb := path.Join(stack.InstanceDir(), "wasm")
+		if dirExists(wasmDb) && !config.Init.ExtractWasm {
+			// By default, remove the wasm DB because it contains native executables.
+			// Hence, it is a security concern when downloading a snapshot from a
+			// untrusted source.
+			err := os.RemoveAll(wasmDb)
```

Comment:
We should also try to remove this directory if extract.Archive fails, in case extracting this part of the archive succeeded. In addition, in both cases, we shouldn't error if the wasm directory simply doesn't exist (check with `errors.Is(err, fs.ErrNotExist)`)

Reply:
> We should also try to remove this directory if extract.Archive fails, in case extracting this part of the archive succeeded.

Makes sense.

> we shouldn't error if the wasm directory simply doesn't exist

We check whether the dir exists before entering the if.